src/app/utils/chart-utils.ts

```diff
@@ -34,3 +35,31 @@ export const chartDurationToDaysMap = {
 }
 
 export const chartUseQueryStaleTimeMs = durationToQueryParams[ChartDuration.TODAY].bucket_size_seconds * 1000
+
+type Buckets = TxVolume[] | undefined
+type MonthlyTxVolume = TxVolume & { numberOfDays: number }
+
+const groupBucketsByMonth = (buckets: Buckets) => {
+  return buckets?.reduce((acc: MonthlyTxVolume[], cur, index, arr) => {
+    if (index > 0 && isSameMonth(new Date(cur.bucket_start), new Date(arr[index - 1].bucket_start))) {
+      acc[acc.length - 1].tx_volume += cur.tx_volume
+      acc[acc.length - 1].numberOfDays += 1
+      return acc
+    }
+    acc.push({
+      bucket_start: cur.bucket_start,
+      tx_volume: cur.tx_volume,
+      numberOfDays: 1,
+    })
+    return acc
+  }, [])
+}
+
+export const getMonthlyBucketsDailyAverage = (buckets: Buckets): Buckets => {
```

Comment:
Not sure why this is needed. Number of days in the month could be calculated from https://date-fns.org/v2.29.3/docs/getDaysInMonth. So that above reduce could be removed. Or am I missing something?

Reply:
Right, disregard. I thought they were full months.